shpkpr/vault.py

```diff
@@ -18,7 +18,7 @@ def resolve_secrets(vault_client, rendered_template):
     for name, definition in secrets.items():
         # parse the secret source and retrieve from vault
         path, key = definition["source"].split(":")
-        path = "secret/{0}".format(path)
+        path = path if path.startswith("secret/") else "secret/{0}".format(path)
```

Comment:
❓ Maybe this isn't completely relevant to this PR, but presumably, we're not limited to storing everything under the `secret` key in vault. Is there a reason that `shpkpr` has opinions about this at all?

Reply:
I think we should delete this logic entirely IMO :+1:

@jeremybini You can have multiple engines https://www.vaultproject.io/intro/getting-started/secrets-engines.html

Reply:
@AlanFoster 👍 makes sense to me. Seems more likely to cause unexpected behavior than it is useful, though admittedly I'm pretty ignorant of potential downstream ramifications 🤷‍♂️.

Reply:
I see this as a transition issue.   If we only support vault full path now, we will have to update all the applications that use that code path to use full path all at once.   By supporting both full and relative path, we can gradually migrate all the application secrets in the near future.   When all our apps are using full path, we can create version 4.1.0 to abandon the support of relative path.

Reply:
@wai-kuen-shopkeep Although that approach would work for ShopKeep's own internal use, as shpkpr is in the public domain this transition should therefore be handled differently. It is ShopKeep's own burden to update to the latest version of shpkpr internally.

In this particular case it seems better to remove this functionality from Shpkpr directly and release a version 4, as it is a backwards incompatible change. ShopKeep can then migrate their own internal tooling to use the latest version as appropriate.

As an aside; the suggestion of releasing 4.0.0, followed by 4.1.0 unfortunately doesn't follow the best practice of [semver](https://semver.org/) - as this behaviour removal is a backwards incompatible change. You would have to release a 4.0.0 followed by 5.0.0

That's my two cents 💰

Reply:
@alan I understand your idea of only support vault full path in v4.0.0.   Does it apply to shpkpr only because public?   I am asking because I am using the same logic in shared-pipeline and marathon-vault-plugin.

Assuming you were only referring to shpkpr, if we only support vault full path going forward, how are we going to transition all the apps to use full path in a timely manner?   The only way I can think of to achieve this is to fork shpkpr to a new repo shpkpr-private.   The new private repo will be used just for the duration of the transition period and it will support both relative and full path.

If you have a better way to handle the transition, please let me know. 

Noted about v5.0.0 instead v.4.1.0.